lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java

```diff
@@ -83,6 +99,8 @@ public static ObjectCannedACL getCannedAcl(S3FileSystemConfig.ObjectCannedAcl ca
     private HostAndPort httpProxy;
     private boolean httpProxySecure;
     private ObjectCannedAcl objectCannedAcl = ObjectCannedAcl.NONE;
+    private RetryMode retryMode = RetryMode.LEGACY;
+    private int maxErrorRetries = 10;
```

Comment:
Did you inspire yourself from hadoop s3 code with the number of retries?

Reply:
yes